### ISSUE TITLE
[BUG REPORT] New repo creation is unfriendly

### ISSUE THREAD

**Issue by elistevens** [tensorwerk/hangar-py#93]
**Describe the bug**

The message `HANGAR RUNTIME WARNING: no repository exists at /some/path/__hangar, please use init_repo function` makes me think my script is doing something wrong, even though the next thing that I do is call `repo.init()`.

Additionally, if the path does not exist, there should be an option to have it be created.

A single default param `exists=True` flag could handle both cases, creating the directory and suppressing the init warning when set to `False`

**Severity**

Select an option:
- [ ] Data Corruption / Loss of Any Kind
- [x] Unexpected Behavior, Exceptions or Error Thrown
- [ ] Performance Bottleneck

**Comment by rlizzo** [comment]
Thanks for the report @elistevens. there are still a few warnings/log messages which really need to be removed now that the project is approaching a more polished state. We'll add it to the todo list before 0.2

**Comment by elistevens** [comment]
It's not particularly urgent, I don't think.

**Comment by rlizzo** [comment]
On another note Eli, what do you think about us naming the hangar folder with two underscores rather than a dot. `/Foo/for/__hangar` rather than `foo/dir/.hangar` 

I'd like it to have the hidden one of the property and general sense of "don't go in this folder unless you know what you're doing" that a `"."folder"` provides, but considering that the amount of data contained in these folders can grow quite significant I didn't want to have somebody forgot about a random folder filling up their hard disk for no reason. 

Also do you think we should put some sort read me in that top level folder explaining don't go in the hangar dir? Along with links to the project

**Comment by elistevens** [comment]
I think that should be driven entirely by a clearly communicated expected use case. I was surprised that there was an extra directory level at all, but that's due to my expectations not being set properly.

Here's an example expected use case that I think could make sense (but is by no means the only one that would):

- dir named `.hangar/`
- lives next to `.git/`
- adds a line to `.gitignore` to ignore `.hangar` (creates file if it doesn't exist)

That would look a bit odd on a global NFS mount, but it would still work. Alternatively, you could have some other expected use case, and you'd tailor it to match.

I think that I'd lean towards getting rid of the intermediate directory entirely, and refuse to init a repo in a non-empty directory. Then the user can put it wherever (`/path/to/git/working/copy/.hangar/` if they like).  But that's based entirely on my weak understanding of what the intended use is.

**Comment by rlizzo** [comment]
So Im currently at the point where u consider the intermediate directory as definitely being important. The original idea was to model it like git. Keeping the code and data together makes a lot of sense intuitively. Sure, larger installations can do whatever they want, but to get started All you need to do is point the hangar client to you projects directory, and go. People understand that workflow. 

It's also nice for CLI usage, since you don't have to navigate into the hangar repo, but instead operate at a level higher where you are less likely to be tempted to play around with files or inadvertently mess stuff up. I know we could solve this issue through static profiles (like `kubectl`) or some other mechanism, but the idea was to make a similar enough to get where people would just understand what to do. 

The hangar repo folder structure is fairly sensitive to change. If you go around and start either moving files or writing data, things are going to really mess up.  At this point we would definitely notice that things are out of place / corrupted, (and it swill make you aware, though the stacktrace would be a mess) but there's not much recovery built in which accounts for user actions, only for the actions possible by the hangar core. My intention is to make it very clear that if you go inside and start making changes you do so at your own risk (which is why I like the ` "."` folders so much.)

For a NFS mounted repo, I guess it would look a bit odd, but it's not a use case I ever really considered. The server directory is named something different, so there's that atleast? But if you wanted to mount a real client repository on NFS I think you might just have to deal with it occurring the same way as git? It seems like an unnecessary bit of complexity to handle how someone wants their data container (essentially a database) named in their system. 

I like the idea about modifying the `.gitconfig` file though! 

Am I missing some big use case which would benefit from this?

**Comment by rlizzo** [comment]
It might be helpful to have a file written in the container director on initialization which basically say

```
This directory has been used to initialize a Hangar Repository

*Foo user* created this Repository on *bar-date* at *time* with and initial version *vX.Y.Z* of hangar. 

Do not modify the contents of the ".hangar" folder under any circumstances. The contents are not meant to be understood by humans. Doing so will result in data loss / corruption. 

More information can be found at *website*

This readme file can be safely modified or deleted at any time. Doing so will not affect the hangar Repository in any way.
```

**Comment by elistevens** [comment]
`.hangar.README.md` IMO.

**Comment by rlizzo** [comment]
This has been addressed on #101! Thanks @elistevens!
